abstract_specification.md

```diff
@@ -233,68 +298,298 @@ Any given storage "engine" upon which SOMA is implemented may have additional fe
+- The remaining parameters are passed directly to the respective type's `create` static method, except for `context`, which is always set to the current collection's context.
+
+`add_new_collection` has an extra parameter allowing control over the type of collection added:
+
+```
+add_new_collection(string key, CollectionType type, string uri = "", PlatformConfig platform_config) -> SOMACollection
+```
+
+- type: The type of collection to add. For instance, if `SOMAExperiment` is provided, the newly-added collection, and the returned instance, will be a `SOMAExperiment`.
 
 ## SOMADataFrame
 
 > ⚠️ **To be further specified** -- all methods need specification.
 >
 > Summary of operations:
 
-| Operation                               | Description                                                                    |
-| --------------------------------------- | ------------------------------------------------------------------------------ |
-| create(uri, ...)                        | Create a SOMADataFrame.                                                        |
-| delete(uri)                             | Delete the SOMADataFrame specified with the URI.                               |
-| exists(uri) -> bool                     | Return true if object exists and is a SOMADataFrame.                           |
-| get soma_type                           | Returns the constant "SOMADataFrame"                                           |
-| get schema -> Arrow.Schema              | Return data schema, in the form of an Arrow Schema                             |
-| get index_column_names -> [string, ...] | Return index (dimension) column names.                                         |
-| get count -> int                        | Return the number of rows in the SOMADataFrame.                                |
-| read                                    | Read a subset of data from the SOMADataFrame                                   |
-| write                                   | Write a subset of data to the SOMADataFrame                                    |
+| Operation                                | Description                                         |
+| ---------------------------------------- | --------------------------------------------------- |
+| static create(uri, ...) -> SOMADataFrame | Create a SOMADataFrame.                             |
+| get soma_type                            | Returns the constant "SOMADataFrame".               |
+| get schema -> Arrow.Schema               | Return data schema, in the form of an Arrow Schema. |
+| get index_column_names -> [string, ...]  | Return index (dimension) column names.              |
+| get count -> int                         | Return the number of rows in the SOMADataFrame.     |
+| read                                     | Read a subset of data from the SOMADataFrame.       |
```

Comment:
`Read a subset of data iteratively from the SOMADataFrame`?  Whether or not one is reading a subset, the iterative nature of reads seems like a core part of the design, even at the spec level, given the returns type for DataFrame.read (and SparseNDArray.read())

While most people bothering to read a spec like this will understand that a subset includes the whole set, do we want to rely upon reader knowing that? "Read all or a subset..."